unixfs.ts

```diff
@@ -0,0 +1,481 @@
+/**
+ * Logical representation of a file chunk.
+ * 
+ * TODO: Clarify when this represenation is used as opposed to `FileChunk`.
+ */
+export interface RawChunk extends PBNode {
+  Data: ByteView<{
+    Type: DataType.Raw,
+    /**
+     * Raw bytes of the content
+     */
+    Data: Bytes
+  }>
+  /**
+   * Raw nodes MUST not have any links, yet empty `Links` list is expected.
+   * At the type level it is expressed as `never[]` which guarantees that
+   * no instatiation other than empty will satisfy this constraint
+   */
+  Links: never[]
+}
+
+/**
+ * Logical representation of a file chunk. When large file is added to IPFS
+ * it gets chunked into smaller pieces (according to specified `--chunker`)
+ * and each chunk is encoded into this representation (and linked from file
+ * DAG).
+ * 
+ * Note: While technically it is possible to add `mode` and `mtime` to
+ * `FileChunk` node it is logical nonsense and therefor to be ignore.
+ */
+export interface FileChunk extends PBNode {
+  Data: ByteView<{
+    Type: DataType.File,
+    /**
+     * Raw bytes corresponding to this file chunk
+     */
+    Data: Bytes
+    /**
+     * Number of bytes in Data field
+     */
+    filesize: uint64
+  }>
+  /**
+   * Raw nodes MUST not have any links, yet empty `Links` list is expected.
+   * At the type level it is expressed as `never[]` which guarantees that
+   * no instatiation other than empty will satisfy this constraint
+   */
+  Links: never[]
```

Comment:
Is this correct here?

Reply:
> Is this correct here?

Comment about types is accurate, however I'm not sure if the claim itself that it should be empty is accurate. According to dag-pb spec `Links` should be present even if empty as per:
https://ipld.io/specs/codecs/dag-pb/spec/#constraints

> Links:
>
>    - must be present, even if empty; the binary form makes no distinction between an empty array and an omitted value, in the Data Model we always instantiate an array.
>    - elements must be sorted in ascending order by their Name values, which are compared by bytes rather than as strings.
>        - Names must be unique or be omitted.

If raw blocks can actually have links, would be great to point that out and also elaborate and provide an example when / what to do with those.

Reply:
the reason for this is to do with the ambiguity of present and not-present in the translation between PB and the data model—you just can't distinguish; so the spec makes it just one way, always present, so there is only one data model form for any given PB form, and not multiple data model forms that can be turned into the same PB bytes.

optionality here doesn't save PB bytes btw, it's the same either way, this is just about 1:1 between in-memory representation and over-the-wire bytes.